### PR TITLE
Bump CBMC submodule and prettify JSON output

### PR DESCRIPTION
**Pull request by chrisr-diffblue**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,4 @@
 [submodule "lib/cbmc"]
 	path = lib/cbmc
+	branch = develop
```
Comment (martin-cs) [inline .gitmodules]: Will this track `develop` ?

File: gnat2goto/ireps/irep_specs/c_enum_member.json [hunk 2]
```diff
@@ -2,6 +2,6 @@
   "namedSub": {
     "value": {"type": "string"},
     "identifier": {"type": "string"},
-    "base_name": {"type": "string"}
+    "baseName": {"type": "string"}
```
Comment (martin-cs) [inline gnat2goto/ireps/irep_specs/c_enum_member.json]: I appreciate that this is probably needed but I wonder why this was changed in CPROVER as minor cosmetic changes to machine <-> machine formats seem a bit daft.

File: testsuite/gnat2goto/tests/string_literal/string_literal.adb [hunk 3]
```diff
@@ -1,5 +1,8 @@
 procedure string_literal is
    A : constant String := "Hello";
+   B : constant String := "Hello";
+   C : constant String := "Goodbye";
 begin
-   null;
+   pragma Assert(A = B);
+   pragma Assert(A /= C);
```
Comment (martin-cs) [inline testsuite/gnat2goto/tests/string_literal/string_literal.adb]: This commit seems unrelated.  I'm OK with it I just wonder if it would be happier in a different PR.

File: gnat2goto/driver/tree_walk.adb [hunk 4]
```diff
@@ -1394,8 +1394,8 @@ package body Tree_Walk is
 
    function Do_String_Constant (N : Node_Id) return Irep is
       Ret              : constant Irep := New_Irep (I_String_Constant_Expr);
-      Element_Type_Ent : constant Entity_Id := Get_Array_Component_Type (N);
-      Element_Type     : constant Irep := Do_Type_Reference (Element_Type_Ent);
+     -- Element_Type_Ent : constant Entity_Id := Get_Array_Component_Type (N);
+     -- Element_Type  : constant Irep := Do_Type_Reference (Element_Type_Ent);
       StrLen           : constant Integer :=
                                     Integer (String_Length (Strval (N)));
       String_Length_Expr : constant Irep := New_Irep (I_Constant_Expr);
```
Comment (xbauch) [inline gnat2goto/driver/tree_walk.adb]: Is this used anywhere?

File: testsuite/gnat2goto/tests/string_literal/string_literal.adb (hunk 3 above)
Comment (chrisr-diffblue) [reply testsuite/gnat2goto/tests/string_literal/string_literal.adb]: It could be in a new PR (happy to do that if you'd prefer) - but I had to tweak string literals a bit to make the submodule bump still pass this regression test, and in doing so I was able to make this test actually do some thing semi-useful (still more work to do to have string literals working properly though).

File: gnat2goto/driver/tree_walk.adb (hunk 4 above)
Comment (chrisr-diffblue) [reply gnat2goto/driver/tree_walk.adb]: Technically not, at the moment - but I think it *will* be again when we finally get types for string literals sorted out. The gnat compiler is happy enough for the moment though as String_Length_Expr gets passed into Set_Type/Set_Value. Happy to comment out these as well if people prefer.

Comment (chrisr-diffblue) [reply gnat2goto/driver/tree_walk.adb]: I've commented out this currently unused code too.

File: .gitmodules (hunk 1 above)
Comment (chrisr-diffblue) [reply .gitmodules]: @martin-cs Documentation to cover the submodule is in https://github.com/diffblue/gnat2goto/pull/119

File: gnat2goto/ireps/irep_specs/c_enum_member.json (hunk 2 above)
Comment (martin-cs) [reply gnat2goto/ireps/irep_specs/c_enum_member.json]: AHHHHHH.... makes sense....

**Comment by martin-cs** [comment]
All fine with me.

**Comment by martin-cs** [comment]
Correction : please check the CI results; esp. the unit tests for packages.

**Comment by chrisr-diffblue** [comment]
@martin-cs - I'm not seeing any test failures other than things that are *XFAILED*:

```
Summary: Ran 42/42 tests, with 0 failed, 0 crashed, 0 unexpectedly passed.
The command "(cd testsuite/gnat2goto; ./testsuite.py --timeout 60 --diffs --enable-color -j 2 )" exited with 0.
0.01s$ gnat2goto/install/bin/unit_tests
##########################################################
# Running suite: Floating point literals
##########################################################
Running test: Integer [Success]
Running test: Between 0 and 1 [Success]
Running test: General number [Success]
Running test: Negative Integer [Success]
Running test: Negative between 0 and 1 [Success]
Running test: General negative number [Success]
Running test: Zero [Success]
All 7 tests succeeded
##########################################################
# Running suite: Binary to hex
##########################################################
Running test: Converting 0 [Success]
Running test: Converting 2 [Success]
Running test: Converting 17 [Success]
Running test: Converting 138 [Success]
Running test: Converting 48879 [Success]
All 5 tests succeeded
The command "gnat2goto/install/bin/unit_tests" exited with 0.
```

**Comment by chrisr-diffblue** [comment]
This is the CI build from before the submodule bump: https://travis-ci.org/diffblue/gnat2goto/builds/485435139

**Comment by martin-cs** [comment]
OK; these two:
https://travis-ci.org/diffblue/gnat2goto/builds/485435139
https://travis-ci.org/diffblue/gnat2goto/builds/487004933

so I will merge.
